src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -142,6 +185,27 @@ fun inferExpressionType(expr: RsExpr): Ty {
     }
 }
 
+private fun tyFromAbsolutePath(prefixStd: String, prefixNoStd: String, name: String, elementForModule: RsCompositeElement): Ty {
+    val module = elementForModule.module ?: return TyUnknown;
+    val crateRoot = elementForModule.crateRoot as? RsFile ?: return TyUnknown
+    val prefix = if (crateRoot.attributes == RsFile.Attributes.NONE) prefixStd else prefixNoStd;
+    val fullName = prefix + "::" + name;
+
+    val (element, _) = org.rust.lang.core.resolve.resolveStringPath(fullName, module) ?: return TyUnknown;
+    return (element as? RsTypeBearingItemElement)?.type ?: TyUnknown;
+}
+
+private fun getMoreCompleteType(t1: Ty, t2: Ty): Ty {
```

Comment:
This is kind of limited and hacky but provides a nice type when only the first or second argument of a range have a concrete type.
e.g.
```rust
let len: usize = xxx;
let range = 0..len;
   //^ Range<usize>
```